Feature/Tasky/v1.0.1-ReusableUi: Creating Reusable UI classes

Alex-A-Fit opened the pull request:
## Feature/Tasky/v1.0.1-ReusableUi: Creating Reusable UI classes

## Bug/Feature description
No reusable UI is created Yet.

## Solution description
Created a Colors class file, a Dimensions data class file, and a Typography class file for later reference when creating UI to eliminate redundant code

## Code Validations 
[X] PR is under 400 lines of code Insertion and Deletion  
[X] Files are Formatted and Optimized  
[X] Clean Architecture followed  
[X] No unneccessary comments or TODOs left in codebase  
[ ] Unit Tests Added  
[X] Unit Tests NOT added and explanation why under exemption description  
[ ] UI Tests Added  
[X] UI Tests NOT added and explanation why under exemption description  
[ ] End to End Tests Added  
[X] End to End Tests NOT added and explanation why under exemption description  
[ ] Screenshot of UI added  

## Exemption Definition
Just creating some basic reusable components

## UI Screenshots 
No UI creation to show

## Test Case Screenshots
No Test Cases to show

Alex-A-Fit committed:
Feature/Tasky/v1.0.1-ReusableUi: Created Colors, Dimensions and Typography Files. Added ttf files for typography use

Alex-A-Fit committed:
Feature/Tasky/v1.0.1-ReusableUi: Added a couple reusable buttons and fixed typography error of 17 sp to 16 sp in subheader2 and 25 to 24 sp in header2 line height

philipplackner commented on core_ui/src/main/java/com/alexafit/core_ui/Typography.kt:
```diff
@@ -0,0 +1,97 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.sp
+
+data class Inter(
+    val bold: FontFamily =
+        FontFamily(
+            Font(R.font.inter_regular, FontWeight.Normal)
+        ),
+    val semiBold: FontFamily =
+        FontFamily(
+            Font(R.font.inter_semibold, FontWeight.SemiBold)
+        ),
+    val regular: FontFamily =
+        FontFamily(
+            Font(R.font.inter_regular, FontWeight.Normal)
+        ),
+    val light: FontFamily =
+        FontFamily(
+            Font(R.font.inter_light, FontWeight.Light)
+        )
+)
```
Could this be an `object` so you don't always need to create an instance when referring to font families?

MKiperszmid commented on core_ui/src/main/java/com/alexafit/core_ui/components/IconActionButton.kt:
```diff
@@ -0,0 +1,28 @@
+
+@Composable
+fun IconActionButton(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    isEnabled: Boolean = true,
+    imageVector: ImageVector,
+    contentDescription: Int
```
modifier should be the first optional parameter, but also, the required parameters (imageVector and contentDescription) should be first than the optional parameters

If not, to set an imageVector and contentDescription you have to use named parameters all the time

So I would:

```
imageVector: ImageVector,
contentDescription: Int,
onClick: () -> Unit,
modifier: Modifier = Modifier,
isEnabled: Boolean = true
```
In fact, if you have a lambda, you can set it as last parameter to make it so the Composable is a high-order function

FInal result:

```
imageVector: ImageVector,
contentDescription: Int,
modifier: Modifier = Modifier,
isEnabled: Boolean = true,
onClick: () -> Unit
```


Now you can call:

```
IconActionButton(vector, description) {
// Your onClick logic
}

```

Alex-A-Fit committed:
Feature/Tasky/v1.0.1-ReusableUi: PR comments addressed:

** made data class an obj to not have to initialize every time for font family

** reorganized parameters as suggested with requirements on top, optional at the bottom and onclick as last to make composable a high order function
